test/OpenTelemetry.Tests/Trace/Sampling/OTelEnvSamplerDetectorTests.cs

```diff
@@ -0,0 +1,89 @@
+// <copyright file="OTelEnvSamplerDetectorTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace OpenTelemetry.Trace.Tests;
+
+public class OTelEnvSamplerDetectorTests : IDisposable
+{
+    public OTelEnvSamplerDetectorTests()
+    {
+        Environment.SetEnvironmentVariable(OTelEnvSamplerDetector.OTelTracesSamplerEnvVarKey, null);
+        Environment.SetEnvironmentVariable(OTelEnvSamplerDetector.OTelTracesSamplerArgEnvVarKey, null);
+    }
+
+    public static IEnumerable<object[]> SupportedSamplers =>
+        new List<object[]>
+        {
+            new object[] { "always_on", null, "AlwaysOnSampler" },
+            new object[] { "always_off", null, "AlwaysOffSampler" },
+            new object[] { "traceidratio", null, "TraceIdRatioBasedSampler{1.000000}" },
+            new object[] { "traceidratio", "2", "TraceIdRatioBasedSampler{1.000000}" },
+            new object[] { "traceidratio", "-1", "TraceIdRatioBasedSampler{1.000000}" },
+            new object[] { "traceidratio", "non-a-number", "TraceIdRatioBasedSampler{1.000000}" },
```

Comment:
these are invalid ratio values and should not be supported

Reply:
Lets discuss it in https://github.com/open-telemetry/opentelemetry-dotnet/pull/3635#discussion_r964534247